### PR TITLE
Support for assuming role for schema registry operations when using kafka connect

### PR DESCRIPTION
**Pull request by liambuchanan**
### Context

* I imagine it's relatively common to share a schema registry among multiple AWS accounts within an organization necessitating cross account access to schema registries.
* Glue Registry does not support resource permissions so cross account have to be used to access schema registries.
* There's currently no way to assume cross account roles when using `AWSKafkaAvroConverter`

### Changes

This PR adds two new configuration parameters used in `AWSKafkaAvroConverter`:
* `sts.roleArn` provides a role ARN to be used when using an AWS glue schema registry
* `sts.roleSessionName` provides a role session name to be used when using an AWS glue schema registry

When both parameters are provided the `AWSKafkaAvroConverter` constructs an `StsAssumeRoleCredentialsProvider` to be used when accessing the registry schema.

### CODE REVIEW

File: avro-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/AWSKafkaAvroConverter.java [hunk 1]
```diff
@@ -44,6 +47,8 @@
 @Slf4j
 @Data
 public class AWSKafkaAvroConverter implements Converter {
+    private static final String STS_ROLE_ARN = "sts.roleArn";
```
Comment (mohitpali) [inline avro-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/AWSKafkaAvroConverter.java:50]: Could you please replace the naming from camel casing to dotted.
e.g. "sts.role.arn"

We have learned that some of the Docker conversions assume Kafka standard of having no camel casing. Some external Docker images provide configurations as ALL CAPS with UNDERSCORES and then translate _ to . and CAPS to small.

File: avro-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/AWSKafkaAvroConverter.java [hunk 2]
```diff
@@ -79,15 +84,19 @@ public AWSKafkaAvroConverter(
     public void configure(Map<String, ?> configs, boolean isKey) {
         this.isKey = isKey;
         new AWSKafkaAvroConverterConfig(configs);
-        final String roleArn = configs.get("sts.roleArn") == null
-                ? null : String.valueOf(configs.get("sts.roleArn"));
-        final String roleSessionName = configs.get("sts.roleSessionName") == null
-                ? null : String.valueOf(configs.get("sts.roleSessionName"));
+        final String roleArn = configs.get(STS_ROLE_ARN) == null
+                ? null : String.valueOf(configs.get(STS_ROLE_ARN));
+        final String roleSessionName = configs.get(STS_ROLE_SESSION_NAME) == null
```
Comment (mohitpali) [inline avro-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/AWSKafkaAvroConverter.java:89]: If session name is not provided, should be default to a default session name ?

**Review by mohitpali — CHANGES_REQUESTED** [review]

**Comment by mohitpali** [comment]
Thanks for this PR ! And I am sorry for getting to it late. I have added some comments. It would be great if you could address and resubmit.

**Comment by austiine04** [comment]
Hello, this PR is perfect for a use case am working on, would love to see it merged in and am happy to work on the requested changes for that, but first i have a question.

Would it better to push the fix to this branch/PR or should i create an entirely new branches with all the changes of this PR plus the feedback?

What's the recommended approach here?

**Comment by liambuchanan** [comment]
@mohitpali updated to use dot notation for config params and added a default value for the session name

**Comment by insani4c** [comment]
Hi @mohitpali ,

This plugin does exactly what is needed for cross-account access to an AWS Glue schema registry.
Could this review please get picked up and merged? I've tested this PR and it works just fine.

Thanks

**Review by mohitpali — CHANGES_REQUESTED** [review]

File: avro-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/AWSKafkaAvroConverter.java [hunk 3]
```diff
@@ -39,6 +47,9 @@
 @Slf4j
 @Data
 public class AWSKafkaAvroConverter implements Converter {
+    private static final String STS_ROLE_ARN = "sts.role.arn";
+    private static final String STS_ROLE_SESSION_NAME = "sts.role.session.name";
+    private static final String STS_ROLE_SESSION_NAME_DEFAULT = "schema-registry-kafkaconnect-converter";
```
[the remark this answers is not in the document]
Comment (mohitpali) [reply avro-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/AWSKafkaAvroConverter.java:52]: > Hello, this PR is perfect for a use case am working on, would love to see it merged in and am happy to work on the requested changes for that, but first i have a question.

> Would it better to push the fix to this branch/PR or should i create an entirely new branches with all the changes of this PR plus the feedback?

> What's the recommended approach here?


You can choose whichever approach works for you. A PR that is from a branch on your fork can be merged directly into master with a PR.

**Comment by dafuallah96** [comment]
Hi, would like to know the reason this PR was closed? It would be great to have this feature as this is something thing that can be greatly used for my use case

**Comment by dafuallah96** [comment]
Hi, would like to know the reason this PR was closed? It would be great to have this feature as this is something thing that can be greatly used for my use case
cc: @liambuchanan @mohitpali

**Comment by liambuchanan** [comment]
@dafuallah96 I only closed as I'm no longer using the schema registry and it appeared that it was going to be a bit of a slog to get this merged. Feel free to pick up where I left off.

**Pull request closed without merging**